src/buildercore/core.py

```diff
@@ -392,11 +394,11 @@ def current_ec2_node_id():
 
     Sample value: 'i-0553487b4b6916bc9'"""
 
-    ensure(env.host is not None, "This is supposed to be called with settings for connecting to an EC2 instance")
-    current_public_ip = env.host
+    ensure(env['host'] is not None, "This is supposed to be called with settings for connecting to an EC2 instance")
```

Comment:
difference in behaviour between fabric and threadbare - 'host' may already exist in fabric as it has loads of defaults, threadbare has zero defaults and an empty dict.

change this to `ensure('host' in env and env['host'], ...)`

Reply:
(self-reviewing, ignore me)